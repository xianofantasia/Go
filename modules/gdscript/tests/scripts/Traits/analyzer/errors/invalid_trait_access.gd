trait SomeTrait:
	enum some_enum {
		A,
		B
	}
	static var some_static_var = 0
	static func some_static_func():
		print("some static func")

func test():
	SomeTrait.new()
	SomeTrait.some_static_func()
	SomeTrait.some_static_var
	SomeTrait.some_enum
