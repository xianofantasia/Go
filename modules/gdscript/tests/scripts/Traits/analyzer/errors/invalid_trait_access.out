GDTEST_ANALYZER_ERROR
>> ERROR at line 11: Cannot call traits functions directly, call through using class.
>> ERROR at line 12: Cannot call traits functions directly, call through using class.
>> ERROR at line 13: Cannot access traits members directly, access through using class.
>> ERROR at line 14: Cannot access traits members directly, access through using class.
